src/components/FastMarker.jsx

```diff
@@ -60,7 +60,25 @@ class FastMarker extends Component {
             && chart.dataSet.length
             && stx.mainSeriesRenderer
         ) {
-            const x = stx.pixelFromDate(this.date, chart);
+            const tick_idx = stx.tickFromDate(this.date, chart);
+            let x = stx.pixelFromTick(tick_idx, chart);
+
+            // ChartIQ doesn't support placing markers in the middle of ticks.
```

Comment:
@amin-binary IMO, it's not a good way to draw a marker for a date that doesn't have a tick because in this way the crosshair shows wrong value for the marker. the best way is to add a null tick for that date to masterData.

Reply:
![Screen Shot 2019-07-18 at 12 19 02 PM](https://user-images.githubusercontent.com/2920178/61428784-43209080-a956-11e9-8b2f-966e59eeef33.png)

@morteza-binary Ideally the crosshair should only show up on ticks, not markers.
but it works fine, even now.

Will add another card to remove the crosshair on markers.

Reply:
@amin-binary because `ChartState.scrollChartToLeft` create an entry for nonexist tick in masterData.

Reply:
@morteza-binary as per our discussion, it's now fixed.